lib/broker/rabbit/queue.ts

```diff
@@ -14,50 +16,84 @@ export class Queue {
 	/**
 	 * Makes a subscription to receive events for a certain routingKey
 	 * @param {string} routingKey - name path for the queue. Example: messages.*.all
-	 * @param {Function}  callback - function to execute actions after event recieved
-	 * @returns {Promise<ConsumeMessage>}
+	 * @param {Function}  callback - function to execute actions after event received
+	 * @returns {Promise<Message>} Promise of Message
 	 *
 	 */
-
 	async subscribe(
 		routingKey: string,
-		callback: Function,
-	): Promise<ConsumeMessage> {
-		let baseName: string;
-		const baseNameArr = routingKey.split('.');
-		baseName = baseNameArr.slice(0, 2).join('.');
-		const name = this.queueName(`${baseName}.all`);
-		this.channel.assertQueue(name);
-		this.channel.bindQueue(name, Config.TOPIC_NAME, routingKey);
-
-		return new Promise<ConsumeMessage>(async (resolve, reject) => {
-			await this.channel.consume(name, (event) => {
-				console.log(JSON.parse(String(event?.content)));
-				resolve(event);
+		callback: (event: Event, context: Context) => void,
+	): Promise<Message> {
+		const name = this.queueName(routingKey);
+		const assertedExchange = await this.channel.assertExchange(
+			Config.TOPIC_NAME,
+			'topic',
+			{ durable: true },
+		);
```

Comment:
You are declaring the Topic on the Queue?

Also, what happens if the exchange doesn't exist? It mustn't create a new exchange.

Reply:
You're right, i'm gonna remove the assertExchange call. 
If the exchange doesn't exists it will throw an error